### PR TITLE
Website translation in Spanish

### PR DESCRIPTION
**Pull request by JoseRivera12**
Lesson 2: Catch'em all
 Introduction
 Chapter 1: Addresses
 Chapter 2: Msg.sender
 Chapter 3: Wild Pokemons
 Chapter 4: Init Function

### CODE REVIEW

**Comment by vasa-develop** [comment]
@gzanitti Can you review this one?

File: es/2/addresses.md [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
-Do NOT change the code below. The below code runs the code editor -->
+# Capítulo 1: Direcciones   
+
+En el último capítulo creamos Pokemones. Pero sin entrenadores para poder entrenarlos, nuestro juego seria aburrido. Agreguemos entrenadores al juego. Cada entrenador tendrá un Pokémon por defecto.
+
+Para esto, necesitamos un nuevo tipo de dato: `address` (dirección).
+
+## Direcciones
+
+La cadena de bloques de Ethereum está formada por **accounts**(cuentas), que puedes visualizar como una cuenta bancaria. Una cuenta tiene un saldo de **Ether** (la moneda usada en la cadena de bloques de Ethereum), donde puedes enviar y recibir pagos de Ethereum de otras cuentas, al igual que en una cuenta bancaria puede trasferir dinero a otras cuentas bancarias.
```
Comment (gzanitti) [inline es/2/addresses.md]:
```suggestion
La cadena de bloques de Ethereum está formada por **accounts**(cuentas), que puedes visualizarlas como análogas a cuentas bancarias. Una cuenta tiene un saldo de **Ether** (la moneda usada en la cadena de bloques de Ethereum), y puedes enviar y recibir pagos de Ethereum desde otras cuentas, al igual que una cuenta bancaria puede transferir dinero a otras cuentas bancarias.
```

File: es/2/addresses.md [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
+
+(Esta dirección le pertenece al equipo de Vyper.fun. ¡Si tu disfrutas nuestro trabajo podrías enviarnos algo de Ether! )
+
+Entraremos más a detalle acerca de las direcciones en la siguiente lección, por ahora solo necesitamos saber que una dirección le pertenece a un usuario en específico o un Smart Contract (contrato inteligente).
+
+Por lo cual podemos usarla para identificar a cada entrenador. Cuando un usuario cree un nuevo entrenador en la aplicación, asignaremos al entrenador a la dirección de Ethereum que llamo a la función.
+
+## Mapeo anidado (nested mapping)
+
+Un mapeo anidado se ve como lo siguiente:
```
Comment (gzanitti) [inline es/2/addresses.md]:
```suggestion
Un mapeo anidado se ve de la siguiente manera:
```

File: es/2/addresses.md [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
+```vyper
+# Asigna un profesor a una lista de estudiantes
+teacherToStudents: HashMap[String[32], HashMap[uint256, String[32]]]
+
+@external
+def addStudent(teacherName: String[32], studentName: String[32], rollnumber: uint256):
+    teacherToStudents[teacherName][rollnumber] = studentName
+```
+
+El `teacherToStudents` asigna un maestro (`String[32]`) a otro sub-mapping que representa una lista de estudiantes. Esta lista de estudiantes asigna el número de lista de estudiante (`uint256`) a su nombre (`String[32]`).
```
Comment (gzanitti) [inline es/2/addresses.md]:
```suggestion
El mapping `teacherToStudents` asigna el nombre de un maestro (`String[32]`) a otro sub-mapping que representa una lista de estudiantes. Esta lista de estudiantes asigna el número del estudiante (`uint256`) a su nombre (`String[32]`).
```

File: es/2/addresses.md [hunk 4]
```diff
@@ -1,5 +1,50 @@
...
+
+El `teacherToStudents` se puede completar agregando 2 claves: la primera para `String[32]` (`teacherName`) y la segunda para `uint256` (`rollnumber`). El valor asignado es un `String[32]` (`studentName`).
```
Comment (gzanitti) [inline es/2/addresses.md]:
```suggestion
El mapping `teacherToStudents` se puede completar agregando 2 claves: la primera para `String[32]` (`teacherName`) y la segunda para `uint256` (`rollnumber`). El valor asignado es un `String[32]` (`studentName`).
```

File: es/2/addresses.md [hunk 5]
```diff
@@ -1,5 +1,50 @@
...
+
+## Ponlo a prueba
+
+Primero, para crear un entrenador, necesitamos una estructura 'entrenador'. Luego para almacenar la asignación de un entrenador y su Pokémon, vamos a necesitar usar 3 mappings(mapeo).
```
Comment (gzanitti) [inline es/2/addresses.md]:
```suggestion
Primero, para crear un entrenador, necesitamos una estructura 'entrenador'. Luego para almacenar la asignación de un entrenador y su Pokémon, vamos a necesitar usar 3 mappings.
```

File: es/2/addresses.md [hunk 6]
```diff
@@ -1,5 +1,50 @@
...
+
+1. Crea una estructura `Trainer` con la propiedad `name` (`String[32]`).
+2. Crea un mapeo(mapping),`trainerPokemonCount` con una clave de tipo `address` y un valor de tipo `uint256`.
```
Comment (gzanitti) [inline es/2/addresses.md]:
```suggestion
2. Crea un mapping,`trainerPokemonCount` con una clave de tipo `address` y un valor de tipo `uint256`.
```

File: es/2/addresses.md [hunk 7]
```diff
@@ -1,5 +1,50 @@
...
+3. Crea un mapeo(mapping), `trainerList` con una clave de tipo `address` y un valor de tipo `Trainer`.
```
Comment (gzanitti) [inline es/2/addresses.md]:
```suggestion
3. Crea un mapping, `trainerList` con una clave de tipo `address` y un valor de tipo `Trainer`.
```

File: es/2/init.md [hunk 8]
```diff
@@ -1,5 +1,56 @@
-<!-- Add translation for the following page: https://vyper.fun/#/2/init
-Do NOT change the code below. The below code runs the code editor -->
+# Capítulo 4: Función de inicio (Init Function)
+
+Nuestro contrato creara pokemones aleatorios con los cuales puede luchar un entrenador. Entonces, necesitamos una lista de nombres de pokemones con los cuales peleara un entrenador.
+
+Necesitamos una forma de almacenar una lista de nombres de pokemones una vez inicializamos nuestro contrato. Para ello utilizaremos la función **\_\_init\_\_**.
+
+## The \_\_init\_\_ Function
```
Comment (gzanitti) [inline es/2/init.md]:
```suggestion
## La función \_\_init\_\_
```

File: es/2/msg-sender.md [hunk 9]
```diff
@@ -1,5 +1,57 @@
...
+> Nota: En Vyper, la ejecución de una función siempre debe comenzar con una llamada externa. Un contrato simplemente permanecerá en la cadena de bloques (blockchain) hasta que alguien llame a alguna de sus funciones. Entonces siempre habrá un `msg.sender`.
+
+Aquí hay un ejemplo de cómo utilizar `msg.sender` y actualizar el mapeo:
+
+```vyper
+favoriteNumber: HashMap[address, uint256]
+
+@external
+def setMyNumber(myNumber: uint256):
+    # Actualiza nuestro `favoriteNumber` mapeo para almacenar `myNumber` en `msg.sender`
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
    # Actualiza nuestro mapeo `favoriteNumber` almacenando el valor `myNumber` usando `msg.sender` como clave
```

File: es/2/msg-sender.md [hunk 10]
```diff
@@ -1,5 +1,57 @@
...
+
+@view
+@external
+def whatIsMyNumber() -> uint256:
+    # Recupera el valor almacenado en la dirección del remitente
+    # Sera `0` si el remitente aún no ha llamado a `setMyNumber`
+    return favoriteNumber[msg.sender]
+```
+
+En este simple ejemplo, cualquiera podría llamar `setMyNumber` y almacenar un `uint256` en nuestro contrato, Que estaría asignado a su `address`. Entonces cuando llamen `whatIsMyNumber`, recibirán el `uint256` que esta almacenado.
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
En este simple ejemplo, cualquiera podría llamar `setMyNumber` y almacenar un `uint256` en nuestro contrato, que estaría asignado a su `address`. Entonces cuando llamen `whatIsMyNumber`, recibirán el `uint256` que está almacenado.
```

File: es/2/msg-sender.md [hunk 11]
```diff
@@ -1,5 +1,57 @@
...
+
+Usando `msg.sender` te brinda la seguridad de que en la cadena de bloques de Ethereum — la única manera de que alguien pueda modificar los datos de otra persona seria robando su clave privada asociada con su dirección de Ethereum.
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
Usar `msg.sender` te brinda la seguridad de que en la cadena de bloques de Ethereum — la única manera de que alguien pueda modificar los datos de otra persona seria robando la clave privada asociada con su dirección de Ethereum.
```

File: es/2/msg-sender.md [hunk 12]
```diff
@@ -1,5 +1,57 @@
...
+
+## Pongámoslo a prueba
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
## Ponlo a prueba
```

File: es/2/init.md [hunk 13]
```diff
@@ -1,5 +1,56 @@
...
+owner: address
+
+@internal
+def __init__():
+    self.owner = msg.sender
+```
+
+No puedes hacer llamadas a otro contrato en la función \_\_init\_\_.
+
+## Pongámoslo a prueba
```
Comment (gzanitti) [inline es/2/init.md]:
```suggestion
## Ponlo a prueba
```

File: es/2/msg-sender.md [hunk 14]
```diff
@@ -1,5 +1,57 @@
...
+
+Necesitamos actualizar las funciones del contrato para que podamos crear un entrenador usando la estructura `Trainer` y el mapeo `trainerList`. Esta función también debería llamar la función `_createPokemon` para crear un Pokémon y luego usar el mapeo `trainerToPokemon` para asignar la propiedad de un Pokémon en un entrenador. Finalmente, incrementamos el contador `trainerPokemonCount`.
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
Necesitamos actualizar las funciones del contrato para que podamos crear un entrenador usando la estructura `Trainer` y el mapping `trainerList`. Esta función también debería llamar a la función `_createPokemon` para crear un Pokémon y luego usar el mapeo `trainerToPokemon` para asignar dicho Pokémon al entrenador. Finalmente, incrementaremos el contador `trainerPokemonCount`.
```

File: es/2/msg-sender.md [hunk 15]
```diff
@@ -1,5 +1,57 @@
...
+
+1. Crea un evento, `NewTrainerCreated` con la propiedad: `name` (`String[32]`).
+
+2. Crea una función `external` llamada `createTrainer` que toma 2 parametros: `trainerName` (`String[32]`) y `pokemonName` (`String[32]`).
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
2. Crea una función `external` llamada `createTrainer` que tome 2 parametros: `trainerName` (`String[32]`) y `pokemonName` (`String[32]`).
```

File: es/2/msg-sender.md [hunk 16]
```diff
@@ -1,5 +1,57 @@
...
+
+3. Dentro de cuerpo de la función `createTrainer`, llama a `_createPokemon` pasando `pokemonName` como el parámetro `_name`. Esto regresara un `Pokemon`. Crea una variable llamada `newPokemon` de tipo `Pokemon` y asigna a su valor el Pokémon que regreso `_createPokemon`.
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
3. Dentro de cuerpo de la función `createTrainer`, llama a `_createPokemon` pasando `pokemonName` como el parámetro `_name`. Esta llamada devolverá un `Pokemon`. Crea una variable llamada `newPokemon` de tipo `Pokemon` y asigna a su valor el Pokémon que obtuviste llamando a `_createPokemon`.
```

File: es/2/msg-sender.md [hunk 17]
```diff
@@ -1,5 +1,57 @@
...
+
+4. Dentro del cuerpo de la función `createTrainer`, crea un entrenador usando la estructura `Trainer` con el `name` como `trainerName`. Crea una variable `newTrainer` de tipo `Trainer` y asigna el valor del entrenador creado.
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
4. Dentro del cuerpo de la función `createTrainer`, crea un entrenador usando la estructura `Trainer` con el `name` como `trainerName`. Crea una variable `newTrainer` de tipo `Trainer` y asígnale el entrenador creado.
```

File: es/2/msg-sender.md [hunk 18]
```diff
@@ -1,5 +1,57 @@
...
+
+5. Agrega el nuevo entrenador al mapeo `trainerList` asignando `msg.sender` a `newTrainer`.
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
5. Agrega el nuevo entrenador al mapeo `trainerList` asignando el valor `newTrainer` a la clave `msg.sender`.
```

File: es/2/msg-sender.md [hunk 19]
```diff
@@ -1,5 +1,57 @@
...
+
+6. Agrega al nuevo entrenador y Pokémon al mapeo anidado `trainerToPokemon` agregando 2 claves: la primera como `msg.sender` y la segunda como `self.trainerPokemonCount[msg.sender]` (que actúa como una clave única para cada Pokémon de un entrenador). El valor asignado es `newPokemon`.
```
Comment (gzanitti) [inline es/2/msg-sender.md]:
```suggestion
6. Agrega al nuevo entrenador y al Pokémon al mapeo anidado `trainerToPokemon` agregando 2 claves: la primera como `msg.sender` y la segunda como `self.trainerPokemonCount[msg.sender]` (que actúa como una clave única para cada Pokémon de un entrenador). El valor asignado es `newPokemon`.
```

File: es/2/wild_pokemons.md [hunk 20]
```diff
@@ -1,5 +1,24 @@
-<!-- Add translation for the following page: https://vyper.fun/#/2/wild_pokemons
-Do NOT change the code below. The below code runs the code editor -->
+# Capítulo 3: Pokemones salvajes
+
+Ahora tenemos entrenadores y sus pokemones. Como todo fan de Pokémon sabe los entrenadores salen de viaje y luchan contra pokemones salvajes. Si ellos ganan la batalla, ellos pueden atrapar al Pokémon.
```
Comment (gzanitti) [inline es/2/wild_pokemons.md]:
```suggestion
Ahora tenemos entrenadores y sus pokemones. Como todo fan de Pokémon sabe los entrenadores salen de viaje y luchan contra pokemones salvajes. Si ellos ganan la batalla, pueden atrapar al Pokémon.
```

File: es/2/wild_pokemons.md [hunk 21]
```diff
@@ -1,5 +1,24 @@
...
+
+Creemos un nuevo contrato con el cual podamos luchar contra pokemones salvajes e intentar atraparlos.
+
+## Pongámoslo a prueba
```
Comment (gzanitti) [inline es/2/wild_pokemons.md]:
```suggestion
## Ponlo a prueba
```

**Review by gzanitti — CHANGES_REQUESTED** [review]

**Comment by gzanitti** [comment]
> @gzanitti Can you review this one?

@vasa-develop Done. Take a look and let's merge if you agree :)

**Comment by gzanitti** [comment]
Congrats @JoseRivera12 !
The PR has been merged.
Thanks for your help 😃

**Pull request merged**